lib/core/presentation/pages/scan_qr_code/scan_qr_code_page.dart

```diff
@@ -0,0 +1,250 @@
+import 'package:app/core/application/event/get_event_checkins_bloc/get_event_checkins_bloc.dart';
+import 'package:app/core/application/event/update_event_checkin_bloc/update_event_checkin_bloc.dart';
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/widgets/scanner_error_widget.dart';
+import 'package:app/core/presentation/widgets/common/appbar/lemon_appbar_widget.dart';
+import 'package:app/core/utils/snackbar_utils.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+
+@RoutePage()
+class ScanQRCodePage extends StatelessWidget {
+  const ScanQRCodePage({
+    super.key,
+    required this.event,
+  });
+  final Event event;
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiBlocProvider(
+      providers: [
+        BlocProvider(
+          create: (context) => UpdateEventCheckinBloc(),
+        ),
+      ],
+      child: ScanQRCodePageView(
+        event: event,
+      ),
+    );
+  }
+}
+
+class ScanQRCodePageView extends StatefulWidget {
+  const ScanQRCodePageView({
+    super.key,
+    required this.event,
+  });
+  final Event event;
+
+  @override
+  State<ScanQRCodePageView> createState() => _ScanQRCodePageViewState();
+}
+
+class _ScanQRCodePageViewState extends State<ScanQRCodePageView> {
+  String overlayText = "Please scan QR Code";
+
+  final MobileScannerController controller = MobileScannerController(
+    formats: const [BarcodeFormat.qrCode],
+    autoStart: true,
+  );
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  void onBarcodeDetect(BarcodeCapture barcodeCapture) {
+    final t = Translations.of(context);
+    setState(() {
+      overlayText = barcodeCapture.barcodes.last.displayValue != null
+          ? t.event.scanQR.scanTicketSuccess
+          : t.common.somethingWrong;
+    });
+    context.read<UpdateEventCheckinBloc>().add(
```

Comment:
is this QR page only used for event feature ? if  no, then let make this page receives the callback rather than couple the event feature with the QR page

Reply:
Yep, it's only for event, the updateEventCheckin also the API only for scan QR and update target scan result user as checked in